test/integration/record_linkage.py

```diff
@@ -0,0 +1,211 @@
+    plant_type_error_prob: float = 0.01,
+    construction_type_error_prob: float = 0.01,
+    construction_year_error_prob: float = 0.01,
+    capacity_mean: float = 500.0,
+    capacity_sigma: float = 10.0,
+    capacity_change_prob: float = 0.01,
+    utility_id_error_prob: float = 0.01,
+    utility_id: int = random.randrange(1000),
+):
+    """Generate a random input DataFrame for testing record linkage."""
+
+    generated_df = pd.DataFrame(
+        {
+            "base_plant_name": [default_plant_name] * size,
+            "plant_type": [
+                random.choice(list(PLANT_TYPE_CATEGORIES["categories"].keys()))
+            ]
+            * size,
+            "report_year": list(range(1994, 1994 + size)),
+            "construction_type": [
+                random.choice(list(CONSTRUCTION_TYPE_CATEGORIES["categories"].keys()))
+            ]
+            * size,
+            "capacity_mw": [capacity_mean] * size,
+            "construction_year": [
+                random.randrange(
+                    VALID_PLANT_YEARS["lower_bound"], VALID_PLANT_YEARS["upper_bound"]
+                )
+            ]
+            * size,
+            "utility_id_ferc1": [utility_id] * size,
+        }
+    )
+
+    # Add random edits to plant name
+    generated_df["plant_name_ferc1"] = generated_df["base_plant_name"].apply(
+        _randomly_modify_string
+    )
+
+    # Modify capacity rows based on probability and sigma
+    generated_df["capacity_mw"] = _noisify(
+        generated_df["capacity_mw"],
+        sigma=capacity_sigma,
+        probability=capacity_change_prob,
+    )
+
+    # Modify categorical columns
+    generated_df = _modify_categorical(
+        generated_df,
+        "construction_type",
+        list(CONSTRUCTION_TYPE_CATEGORIES["categories"]),
+        construction_type_error_prob,
+    )
+    generated_df = _modify_categorical(
+        generated_df,
+        "construction_year",
+        list(range(VALID_PLANT_YEARS["lower_bound"], VALID_PLANT_YEARS["upper_bound"])),
+        construction_year_error_prob,
+    )
+    generated_df = _modify_categorical(
+        generated_df,
+        "plant_type",
+        list(PLANT_TYPE_CATEGORIES["categories"]),
+        plant_type_error_prob,
+    )
+
+    # Generate l1 unit vector to represent fuel fractions (all add up to 1)
+    fuel_frac_vec = abs(_RANDOM_GENERATOR.normal(size=len(_FUEL_COLS)))
+    fuel_frac_vec = fuel_frac_vec / np.linalg.norm(fuel_frac_vec, ord=1)
+    generated_df[_FUEL_COLS] = fuel_frac_vec
+
+    # Add minor noise to fuel fractions
+    for col in _FUEL_COLS:
+        generated_df[col] = _noisify(generated_df[col])
+
+    return generated_df
+
+
+@pytest.fixture
+def mock_ferc1_plants_df():
+    """Returns a test DataFrame for use in generic record linkage testing."""
+    # Creates test dataframes for a bunch of plants and concatenates them
+    return pd.concat(
+        [
+            _generate_random_test_df("fox lake, mn"),
```

Comment:
I picked some random names that seemed somewhat representative of the types of plant names we see, and random edits will be applied to certain rows. I think this is a decent reflection of some of the types of errors we see in names, but there's probably other types that this misses